README.md

```diff
@@ -37,30 +38,26 @@ typeof ch // -> 'function'
 ### Sending values to the channel
 
 Values are added to the
-channel by calling the function with either `(value)` or `(error, value)`. The
-return value is a thunk (a function that take a node-style callback as its only
-argument). The callback given to the thunk is called once the value is added.
+channel by calling the function with either `(value)` or `(error, value)`.
+The return value is a Promise.
 
 ```js
-ch('foo')(function (err) {
-  if (err) {
-    // There was an error putting the value on the channel
-  } else {
-    // The value was successfully put on the channel
-  }
+ch('foo').then(function () {
+  // The value was successfully put on the channel
+}).catch(function (err) {
+  // There was an error putting the value on the channel
 })
 ```
 
 ### Receiving values from the channel
 
-Values are removed from the channel by calling it with a node-style callback as
-this first argument. When a value is available on the channel the callback is
-called with the value or error. In this case the channel itself can also be a
-thunk.
+Values are taken off the channel by resolving the channel promise repeatedly.
 
 ```js
-ch(function (err, val) {
-  // called when there is a value or error on the channel
+ch.then(function (val) {
```

Comment:
This doesn't seem to be implemented. The channel is a function does not have a then method.